pkg/controller/constraint/constraint_controller.go

```diff
@@ -201,6 +201,7 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 		log.Error(err, "unpacking request", "request", request)
 		return reconcile.Result{}, nil
 	}
+	defer r.tracker.Observe(gvk.Group, gvk.Kind, unpackedRequest.Name, unpackedRequest.Namespace)
```

Comment:
Same comment here as the config reconciler. I don't believe we want to simply defer a call to `Observe()` because we may want to handle errors and retries, and some times it makes sense to cancel the expectation.